### ISSUE TITLE
Extract an expression as a non-void function

### ISSUE THREAD

**Issue by thoni56** [thoni56/c-xrefactory#28]
If you select an expression, e.g. inside an `if` statement, and extract that as a function you get a `void` function. Either `c-xref` should recognize the text as in the context of an expression, or give you an option to extracting it as a non-void function with a return statement.

The latter is probably the only choice for C since everything is an expression and you can have expressions everywhere. It seems that in Java the distinction is made since there are both _Extract Method_ and _Extract Function_, but only the latter is available for C.

**Comment by shanemd** [comment]
This could possibly be accomplished by combining 'extract variable' on the expression, then applying 'extract' function to the new variable initialization.

I agree that being able to extract expressions in general would be a useful enhancement.

**Comment by thoni56** [comment]
You are right! I haven't tried that. Perhaps that is also the way to implement the feature, by combining those functions.
